aten/src/ATen/native/cuda/AmpKernels.cu

```diff
@@ -0,0 +1,110 @@
+#define _USE_MATH_DEFINES
+
+#include <math.h>
+
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cuda/Loops.cuh>
+
+
+namespace at {
+namespace native {
+// These functions have no CPU variants, so there's no need to bother with a .cpp file in native/,
+// and also no need to impose the DECLARE/DEFINE/REGISTER_DISPATCH layer of indirection.
+
+// _amp_unscale_inf_check_cuda acts in-place on scaled_grad, multiplying each element by rscale's value.
+// If any thread finds an inf/nan, it posts a 1. to found_inf.
+//
+// Args:
+// scaled_grad:  An incoming scaled gradient tensor, which may contain infs/nans
+// rscale:  The inverse of the scale factor by which scaled_grad is currently multiplied.
+// found_inf:  A tensor to record whether scaled_grad contained any infs/nans
```

Comment:
Singleton tensor. Initial value should be 0.

Reply:
It won't necessarily be 0 initially every time this function is entered, because the same found_inf tensor is used to for the inf/nan check for all the parameters.  found_inf will initially be zero when entering this loop over parameters, but might not be zero by the time we reach later parameters.

In general pre-zeroing this tensor is the responsibility of the caller.  Added a note saying so.